clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3643,6 +3643,64 @@ make the function's CFI jump table canonical. See :ref:`the CFI documentation
   }];
 }
 
+def KCFISaltDocs : Documentation {
+  let Category = DocCatFunction;
+  let Content = [{
+Use ``__attribute__((kcfi_salt("<salt>")))`` on a function declaration, function
+definition, or typedef to help distinguish CFI hashes between functions with
+the same type signature.
```

Comment:
Like in the kernel patch, it would be nice to summarize the use case, i.e. in what cases we would want to apply this.

Reply:
I added some of Sami's examples to the docs. Let me know what you think.